src/OpenLoco/Paint/PaintEntity.cpp

```diff
@@ -0,0 +1,99 @@
+#include "PaintEntity.h"
+#include "../Config.h"
+#include "../Interop/Interop.hpp"
+#include "../Map/Tile.h"
+#include "../Things/Misc.h"
+#include "../Things/ThingManager.h"
+#include "Paint.h"
+
+using namespace OpenLoco::Interop;
+using namespace OpenLoco::Ui::ViewportInteraction;
+
+namespace OpenLoco::Paint
+{
+    // 0x004B0CCE
+    void paintVehicleEntities(PaintSession& session, vehicle_base* base)
+    {
+        registers regs{};
+        regs.ax = base->x;
+        regs.cx = base->y;
+        regs.dl = base->z;
+        regs.ebx = (base->sprite_yaw + (session.rotation() << 4)) & 0x3F;
+        regs.esi = reinterpret_cast<int32_t>(base);
+        call(0x004B0CCE, regs);
+    }
+
+    // 0x00440325
+    void paintMiscEntities(PaintSession& session, MiscBase* base)
+    {
+        registers regs{};
+        regs.ax = base->x;
+        regs.cx = base->y;
+        regs.dl = base->z;
+        regs.ebx = (base->sprite_yaw + (session.rotation() << 4)) & 0x3F;
+        regs.esi = reinterpret_cast<int32_t>(base);
+        call(0x00440325, regs);
+    }
+
+    // 0x0046FA88
+    void paintEntities(PaintSession& session, const Map::map_pos& loc)
+    {
+        auto* dpi = session.getContext();
+        if (Config::get().vehicles_min_scale < dpi->zoom_level)
+        {
+            return;
+        }
+
+        if (loc.x >= 0x4000 || loc.y >= 0x4000)
+        {
+            return;
+        }
+
+        ThingManager::ThingTileList entities(loc);
+        for (auto* entity : entities)
+        {
+            auto left = dpi->x;
```

Comment:
Maybe add a todo to convert this to some `rect.contains(otherRect)`-like function

Reply:
Yes i was also thinking of similar.